pkg/kubectl/cmd/apply_test.go

```diff
@@ -279,6 +296,85 @@ func TestApplyObject(t *testing.T) {
 	}
 }
 
+// TestApplyTPR uses apply commad with ThirdPartyResources.
+func TestApplyTPR(t *testing.T) {
+	initTestErrorHandler(t)
+	nameTPR := "test-tpr.unit-test.test.com"
+	pathTPR := "/thirdpartyresources/" + nameTPR
+	currentTPR := readTPRFromFile(t, filenameTPR)
+
+	verifiedPatch := false
+
+	f, tf, _, ns := cmdtesting.NewAPIFactory()
+	tf.Printer = &testPrinter{}
+	tf.JSONEncoder = testapi.Extensions.Codec()
+	tf.UnstructuredClient = &fake.RESTClient{
+		ContentConfig: restclient.ContentConfig{
+			ContentType: runtime.ContentTypeJSON,
+			GroupVersion: &schema.GroupVersion{
+				Group:   extensions.GroupName,
+				Version: "v1beta1",
+			},
+			NegotiatedSerializer: ns,
+		},
+		NegotiatedSerializer: ns,
+		Client: fake.CreateHTTPClient(func(req *http.Request) (*http.Response, error) {
+			switch p, m := req.URL.Path, req.Method; {
+			case p == pathTPR && m == "GET":
+				bodyTPR := ioutil.NopCloser(bytes.NewReader(currentTPR))
```

Comment:
For stubbing return data, I would hardcode the result as much as possible for ease of inspection